### PR TITLE
Report usage of star imports

### PR DESCRIPTION
**Pull request by jayvdb**

### CODE REVIEW

**Comment by jayvdb** [comment]
Happy to split this into two PRs.  I'm basing a few of my pending changes on "Simplify handleNodeLoad scope selection logic".

If this concept is seen as desirable, I'll add more tests for corner cases.

**Comment by bitglue** [comment]
:+1:, looks great. I've added you as a collaborator so you can merge your own stuff, and I don't have to keep asking you to rebase things. Feel free to merge whenever you feel a change is good, the tests pass, and at least one person (doesn't have to be) has reviewed it. Thanks for all your contributions so far!

File: pyflakes/checker.py [hunk 1]
```diff
@@ -147,6 +147,17 @@ def redefines(self, other):
...
 
 
+class StarImportation(Importation):
+    """A binding created by an 'from x import *' statement."""
+
+    def __init__(self, name, source):
+        super(StarImportation, self).__init__('*', source)
+        # Each star importation needs a unique name, and
+        # may not be the module name otherwise it will be deemed imported
+        self.name = name + '.*'
```
Comment (jayvdb) [inline pyflakes/checker.py]: I've improved this changeset considerably, creating a separate StarImportation binding for each import.  This line here is a new 'ugly'.  I havent thought of a better way yet.

File: pyflakes/checker.py [hunk 2]
```diff
@@ -358,17 +369,29 @@ def checkDeadScopes(self):
             if isinstance(scope, ClassScope):
                 continue
 
-            if isinstance(scope.get('__all__'), ExportBinding):
-                all_names = set(scope['__all__'].names)
+            all_binding = scope.get('__all__')
```
Comment (jayvdb) [inline pyflakes/checker.py]: fwiw, this section of code around `*` vs `__all__` has changed quite a bit.

File: pyflakes/test/test_imports.py [hunk 3]
```diff
@@ -628,6 +628,17 @@ class a:
             from fu import *
         ''', m.ImportStarNotPermitted)
 
+    @skipIf(version_info > (3,),
+            'import * below module level is an on Python 3')
```
Comment (bitglue) [inline pyflakes/test/test_imports.py]: Is there a missing word in this statement? "is an" what?

**Comment by bitglue** [comment]
LGTM except for minor inline comment above.
